src/app.rs

```diff
@@ -1462,6 +1463,33 @@ This flag can be disabled with --no-search-zip.
     args.push(arg);
 }
 
+fn flag_preprocessor(args: &mut Vec<RGArg>) {
+    const SHORT: &str = "search outputs of \"COMMAND FILE\" for each FILE";
+    const LONG: &str = long!("\
+For each input FILE, search the standard output of \"COMMAND FILE\" rather
+than the contents of FILE.  This option expects the COMMAND program to be
+available in your PATH.  An empty string COMMAND deactivats this feature.
+
+When searching over sets of files that may require one of several decoders
+as preprocessors, COMMAND should be a wrapper program or script which first
+classifies FILE based on magic numbers/content or based on the FILE name and
+then dispatches to an appropriate preprocessor.  Each COMMAND also has its
+standard input connected to FILE for convenience.
+
+For example, a Unix script for COMMAND might look like:
+case \"$1\" in
+  *.pdf) exec pdftotext \"$1\" - ;;
+  *) case $(file \"$1\") in
+       *Zstandard*) exec pzstd -cdqp8 ;;
+       *) exec cat ;;
+	 esac;
+esac
+");
+    let arg = RGArg::flag("preprocessor", "COMMAND").short("P")
```

Comment:
Usually when one option overrides another (as you said this does with `-z`), you put that in the help text and then specify it here. You can see examples of that elsewhere in the file

I actually would have suggested making them *conflict*, because it's not intuitively obvious how they should behave together, but then i thought that a lot of people probably add `-z` to their aliases/configs, so that might be too inconvenient. Not sure